Package.swift

```diff
@@ -1,29 +1,53 @@
-// swift-tools-version:4.2
+// swift-tools-version:5.1
 
 import PackageDescription
+import Foundation
+
+#if os(Linux)
+import Glibc
+#else
+import Darwin.C
+#endif
+
+// Enforcing exclusivity increases compile time of release builds by 2 minutes.
+// Disable it.
+var swiftSyntaxUnsafeFlags: [String] = [
+  "-enforce-exclusivity=unchecked"
+]
+
+// If building SwiftSyntax as a dylib we are in a controlled CI environment.
+// Thus we can use internal compiler flags like the group info.
+if getenv("SWIFT_SYNTAX_BUILD_DYLIB") != nil {
+  let groupFile = URL(fileURLWithPath: #file)
+    .deletingLastPathComponent()
+    .appendingPathComponent("utils")
+    .appendingPathComponent("group.json")
+  swiftSyntaxUnsafeFlags += ["-Xfrontend", "-group-info-path",
+                             "-Xfrontend", groupFile.path]
```

Comment:
Just curious: What does this do? 🤔

Reply:
This enables grouping the public API in subgroups for interface generation purposes, instead of generating an interface that is a flat list of everything. It's primarily intended for use for the stdlib interface: https://github.com/apple/swift/blob/master/stdlib/public/core/GroupInfo.json

Reply:
aah, neat